modules/querying/pages/select-statement/index.adoc

```diff
@@ -0,0 +1,2866 @@
+    },
+    {
+      "v_id": "person10",
+      "attributes": {
+        "@numCoworkers": 1,
+        "@numFriends": 2,
+        "id": "person10"
+      },
+      "v_type": "person"
+    },
+    {
+      "v_id": "person7",
+      "attributes": {
+        "@numCoworkers": 6,
+        "@numFriends": 1,
+        "id": "person7"
+      },
+      "v_type": "person"
+    },
+    {
+      "v_id": "person5",
+      "attributes": {
+        "@numCoworkers": 5,
+        "@numFriends": 1,
+        "id": "person5"
+      },
+      "v_type": "person"
+    },
+    {
+      "v_id": "person11",
+      "attributes": {
+        "@numCoworkers": 1,
+        "@numFriends": 1,
+        "id": "person11"
+      },
+      "v_type": "person"
+    }
+  ]}]
+}
+----
+--
+====
+
+
+== `LIMIT` Clause
+
+The optional `LIMIT` clause sets constraints on the number and ranking of items included in the final result set.
+
+.EBNF for `LIMIT` Clause
+[source,ebnf]
+----
+limitClause := LIMIT ( expr | expr "," expr | expr OFFSET expr )
+----
+
+Each of the expression must evaluate to a non-negative integer.
+To understand `LIMIT`, note that the tentative result set is held in the computer as a list of vertices.
+If the query has an `ORDER BY` clause, the order is specified; otherwise the list order is unknown.
+Assume we number the vertices as `v_1` , `v_2` , ..., `v_n`.
+The `LIMIT` clause specifies a range of vertices, starting from a lower position in the list to an upper position.
+
+There are three forms:
+
+.`LIMIT` scenarios
+[source.wrap,gsql]
+----
+result = SELECT v FROM S -(:e)- :v LIMIT k; <1>
+result = SELECT v FROM S -(:e)- :v LIMIT j, k; <2>
+result = SELECT v FROM S -(:e)- :v LIMIT k OFFSET j; <3>
+----
+<1> Case 1: k = Count
+<2> Case 2: j = Offset from the start of the list, k = Count
+<3> Case 3: k = Count, j = Offset from the start of the list
+
+Case 1: `LIMIT k`
+
+* When a single expr is provided, `LIMIT` returns the first k elements from the tentative result set.
+If there are fewer than *k* elements available, then all elements will be returned in the result set.
+If k=5 and the tentative result set has at least 5 items, then the final result list will be [ v_1 , v_2 , v_3 , v_4 , v_5 ].
+
+Case 2: `LIMIT j, k`
+
+* When a comma separates two expressions, LIMIT treats the first expression *j* as an offset.  That is, it skips the first *j* items in the list.  The second expr *k* tells the maximum number of items to include. If the list has at least 7 items, then LIMIT 2, 5 would return [ v_3 , v_4 , v_5, v_6  _,_ v_7 ].
+
+Case 3: `LIMIT k OFFSET j`
+
+* The behavior of Case 3 is the same as that of Case 2, except that the syntax is different.  The keyword OFFSET separates the two expressions, and the count comes before the offset, rather than vice versa. If the list has at least 7 items, then LIMIT 5 OFFSET 2 would return [ v_3 , v_4 , v_5, v_6 , v_7 ].
+
+If any of the expressions evaluate to a negative integer, the results are undefined.
```

Comment:
Does this result in an error?

Reply:
I don't think so. The original author says results are undefined - which doesn't mean there would be an error.